contracts/Vault2.sol

```diff
@@ -1,6 +1,24 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.4;
 
-contract Vault2 {
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+contract Vault2 is ERC20 {
+    address public owner;
+
+    constructor() ERC20('Vault2', 'V2'){
+        _mint(msg.sender, 10000);
+        owner = msg.sender;
+    }
+
+    function mint(address to, uint amount) external {
+        require(msg.sender == owner, "Only owner can mint");
```

Comment:
You can remove this, anyone can mint given he supplies an equal amount of ether. so basically you should be doing `_mint(msg.sender, msg.value)` and same with burn